src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/IntegrationTests/ComponentCodeGenerationTestBase.cs

```diff
@@ -19,6 +19,76 @@ protected ComponentCodeGenerationTestBase()
 
         #region Basics
 
+        [Fact]
+        public void ChildComponent_InFunctionsDirective()
+        {
+            // Arrange
+            AdditionalSyntaxTrees.Add(Parse(@"
+using Microsoft.AspNetCore.Components;
+
+namespace Test
+{
+    public class MyComponent : ComponentBase
+    {
+    }
+}
+"));
+
+            // Act
+            var generated = CompileToCSharp(@"
+@addTagHelper *, TestAssembly
+@using Microsoft.AspNetCore.Components.RenderTree;
+
+@{ RenderChildComponent(builder); }
+
+@functions {
+    void RenderChildComponent(RenderTreeBuilder builder)
```

Comment:
@rynowak in order to make Html constructs work in components `@functions` directive you need to pass in the builder. Thoughts? Makes me think there's more value to `@helper`. But then again, if you look below to the local functions test you don't have that restriction.

@ajaybhargavb good call on adding the components tests 👌

Reply:
I wasn't expected we would make components work during this PR 😆 😎